calico-vpp-agent/cni/network_vpp.go

```diff
@@ -350,4 +376,7 @@ func (s *Server) DelVppInterface(podSpec *storage.LocalPodSpec) {
 		Type: common.PodDeleted,
 		Old:  podSpec,
 	})
+	buffersNeededForThisPod := calculateBuffersNeededForPod(podSpec)
+	s.buffersNeeded -= buffersNeededForThisPod
```

Comment:
I'm a bit worried that tracking the buffers used in VPP with s.buffersNeeded may become inaccurate over time (for instance if the annotation for a pod is changed during its lifetime (after the interface is created and before it is deleted). 
It might be better to just get the number of available buffers in vpp each time before creating the interface. @sknat  any opinion on this?

Reply:
yes, cleaner, but higher cost.